pkg/operation/botanist/botanist.go

```diff
@@ -108,3 +112,35 @@ func (b *Botanist) RequiredExtensionsReady(ctx context.Context) error {
 
 	return nil
 }
+
+// CreateETCDSnapshot executes to the ETCD main Pod and triggers snapshot.
+func (b *Botanist) CreateETCDSnapshot(ctx context.Context) error {
+	executor := kubernetes.NewPodExecutor(b.K8sSeedClient.RESTConfig())
+	namespace := b.Shoot.SeedNamespace
+	etcdMainSelector := getETCDMainLabelSelector()
+
+	podsList := &v1.PodList{}
+	if err := b.K8sSeedClient.Client().List(ctx, podsList, client.InNamespace(namespace), client.MatchingLabelsSelector{Selector: etcdMainSelector}); err != nil {
+		return err
+	}
+
+	if podsList == nil {
```

Comment:
I don't think `podList` can be nil here, perhaps you meant an empty Items list?
```suggestion
	if len(podsList.Items) == 0 {
```